lib/Doctrine/MongoDB/Event/DistinctEventArgs.php

```diff
@@ -47,18 +47,45 @@ public function __construct($invoker, $field, array $query)
         $this->query = $query;
     }
 
+    /**
+     * @return object
+     */
     public function getInvoker()
     {
         return $this->invoker;
     }
 
+    /**
+     * @return string
+     */
     public function getField()
     {
         return $this->field;
     }
 
+    /**
+     * @return array
+     */
     public function getQuery()
     {
         return $this->query;
     }
+
+    /**
+     * @param $query
+     * @since 1.3
+     */
+    public function setQuery($query)
+    {
+        $this->query = $query;
+    }
+
+    /**
+     * @param string $field
+     * @since 1.3
+     */
+    public function setField($field)
+    {
+        $this->field = $field;
```

Comment:
@blockjon: I realize the constructor is outside the scope of your PR, but in the interest of defensive programming it might make sense to perform `(string)` casting whenever we assign the `$field` property (here and in the constructor).